src/Command/DrupalUpdate/UpdateScriptUtility.php

```diff
@@ -0,0 +1,317 @@
+<?php
+
+
+namespace Acquia\Cli\Command\DrupalUpdate;
+
+use PharData;
+
+class UpdateScriptUtility
+{
+  private $ignore_updates_files;
+
+  /**
+   * UpdateScriptUtility constructor.
+   */
+  public function __construct() {
+    $this->ignore_updates_files = [
+          '.gitignore','.htaccess','CHANGELOG.txt','sites',
+      ];
+  }
+
+  /**
+   * Generate tabular view of available updates
+   * @param $version_detail
+   */
+  function printReleaseDetail($version_detail) {
+    $string_count = $this->getStringCountOfColumn($version_detail);
+    foreach ($version_detail as $key => $value){
+      echo str_repeat("-", array_sum($string_count)+6);
+      echo "\n";
+      echo "| ";
+      $i=0;
+      foreach ($value as $v){
+        if(is_array($v)){
+          foreach ($v as $t => $a){
+            echo $a . ",";
+          }
+          echo " | ";
+        } else{
+          $repetor = $string_count[$i]-strlen($v);
+          if($repetor > 0){
+            echo $v . str_repeat(" ", $string_count[$i]-strlen($v)) . " | ";
+          }else{
+            echo $v . str_repeat(" ", 1) . " | ";
+          }
+
+        }
+        $i++;
+      }
+      echo "\n";
+    }
+    echo "\n";
+  }
+
+  /**
+   * Update code based on available security update.
+   * @param $latest_security_updates
+   */
+
+  function updateCode($latest_security_updates) {
+    foreach ($latest_security_updates as $k => $value){
+      if(!isset($value['download_link'])){
+        continue;
+      }
+      if($value['package']=='drupal'){
+        $dirname = 'temp_drupal_core';
+        $filename = $value['file_path'] . "/" . $dirname . "";
+        if (!file_exists($filename)) {
+          $oldmask = umask(0);
+          mkdir($value['file_path'] . "/" . $dirname, 0777);
```

Comment:
Please use Symfony Filesystem for all filesystem operations https://symfony.com/doc/current/components/filesystem.html